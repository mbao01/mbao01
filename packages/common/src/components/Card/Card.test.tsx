import { render, screen } from "@testing-library/react";
import { Card } from "./";

describe("Card", () => {
  it("renders a basic card", () => {
    const { asFragment } = render(
      <Card>
        <Card.Content>
          If a dog chews shoes whose shoes does he choose?
        </Card.Content>
      </Card>
    );

    expect(
      screen.getByText("If a dog chews shoes whose shoes does he choose?")
    ).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a card with a header", () => {
    const { asFragment } = render(
      <Card>
        <Card.Content>
          <Card.Header>Nike Shoe!</Card.Header>
          Real cool shoe
        </Card.Content>
      </Card>
    );

    expect(
      screen.getByRole("heading", { name: "Nike Shoe!", level: 3 })
    ).toBeInTheDocument();
    expect(screen.getByText("Real cool shoe")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a card with header and footer", () => {
    const { asFragment } = render(
      <Card>
        <Card.Content>
          <Card.Header>Nike Shoe!</Card.Header>
          Real cool shoe
          <Card.Footer>Some footer content</Card.Footer>
        </Card.Content>
      </Card>
    );

    expect(
      screen.getByRole("heading", { name: "Nike Shoe!", level: 3 })
    ).toBeInTheDocument();
    expect(screen.getByText("Real cool shoe")).toBeInTheDocument();
    expect(screen.getByText("Some footer content")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a card with a description", () => {
    const { asFragment } = render(
      <Card>
        <Card.Content>
          <Card.Header>Nike Shoe!</Card.Header>
          <Card.Description>Some real cool shoe</Card.Description>
        </Card.Content>
      </Card>
    );

    expect(
      screen.getByRole("heading", { name: "Nike Shoe!", level: 3 })
    ).toBeInTheDocument();
    expect(screen.getByText("Some real cool shoe")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a card with image", () => {
    const { asFragment } = render(
      <Card>
        <Card.Content>Real cool shoe</Card.Content>
        <Card.Image src="image.jpg" alt="my image" />
      </Card>
    );

    expect(screen.getByText("Real cool shoe")).toBeInTheDocument();
    expect(screen.getByRole("img", { name: "my image" })).toHaveAttribute(
      "src",
      "image.jpg"
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a card with border", () => {
    const { asFragment } = render(
      <Card bordered>
        <Card.Content>Real cool shoe</Card.Content>
      </Card>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a card with overlay", () => {
    const { asFragment } = render(
      <Card bordered>
        <Card.Content>Real cool shoe</Card.Content>
        <Card.Image src="image.jpg" alt="my image" />
      </Card>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a horizontal card with image on the left", () => {
    const { asFragment } = render(
      <Card horizontal>
        <Card.Content>Real cool shoe</Card.Content>
        <Card.Image src="image.jpg" alt="my image" />
      </Card>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a compact card", () => {
    const { asFragment } = render(
      <Card compact>
        <Card.Content>Real cool shoe</Card.Content>
      </Card>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
