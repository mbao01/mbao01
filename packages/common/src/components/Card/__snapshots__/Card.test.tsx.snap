// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > renders a basic card 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      If a dog chews shoes whose shoes does he choose?
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card > renders a card with a description 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <h3
        class="card-title"
      >
        Nike Shoe!
      </h3>
      <div
        class="text-sm text-muted-foreground"
      >
        Some real cool shoe
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card > renders a card with a header 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <h3
        class="card-title"
      >
        Nike Shoe!
      </h3>
      Real cool shoe
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card > renders a card with border 1`] = `
<DocumentFragment>
  <div
    class="card card-border"
  >
    <div
      class="card-body"
    >
      Real cool shoe
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card > renders a card with header and footer 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      Nike Shoe!
    </div>
    <div
      class="p-6 pt-0"
    >
      Real cool shoe
    </div>
    <div
      class="flex items-center p-6 pt-0"
    >
      Some footer content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card > renders a card with image 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      Real cool shoe
    </div>
    <figure>
      <img
        alt="my image"
        src="image.jpg"
      />
    </figure>
  </div>
</DocumentFragment>
`;

exports[`Card > renders a card with overlay 1`] = `
<DocumentFragment>
  <div
    class="card card-dash"
  >
    <div
      class="card-body"
    >
      Real cool shoe
    </div>
    <figure>
      <img
        alt="my image"
        src="image.jpg"
      />
    </figure>
  </div>
</DocumentFragment>
`;

exports[`Card > renders a card with title and actions 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <h3
        class="card-title"
      >
        Nike Shoe!
      </h3>
      Real cool shoe
      <div
        class="card-actions"
      >
        Some actions content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card > renders a extra large (xl) card 1`] = `
<DocumentFragment>
  <div
    class="card card-xl"
  >
    <div
      class="card-body"
    >
      Real cool extra large shoe
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card > renders a horizontal card with image on the left 1`] = `
<DocumentFragment>
  <div
    class="card card-side"
  >
    <div
      class="card-body"
    >
      Real cool shoe
    </div>
    <figure>
      <img
        alt="my image"
        src="image.jpg"
      />
    </figure>
  </div>
</DocumentFragment>
`;

exports[`Card > renders a large (lg) card 1`] = `
<DocumentFragment>
  <div
    class="card card-lg"
  >
    <div
      class="card-body"
    >
      Real cool large shoe
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card > renders a medium (md) card 1`] = `
<DocumentFragment>
  <div
    class="card card-md"
  >
    <div
      class="card-body"
    >
      Real cool medium shoe
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card > renders a small (sm) card 1`] = `
<DocumentFragment>
  <div
    class="card card-sm"
  >
    <div
      class="card-body"
    >
      Real cool small shoe
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card > renders a tiny (xs) card 1`] = `
<DocumentFragment>
  <div
    class="card card-xs"
  >
    <div
      class="card-body"
    >
      Real cool tiny shoe
    </div>
  </div>
</DocumentFragment>
`;
